File: myshoppinglistapi/src/main/java/com/example/myshoppinglistapi/entities/Purchase.java
package com.example.myshoppinglistapi.entities;

import javax.persistence.*;

public class Purchase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "idPurchase")
    private Long id = 0L;

    @Column(name = "name")
    private String name = "";

    @Column(name = "locale")
    private String locale = "";

    @Column(name = "quantiOrKilo")
    private String quantiOrKilo = "";

    @Column(name = "typeProduct")
    private int typeProduct = 0;

    @Column(name = "date")
    private String date = "24-01-2022";

    @Column(name = "price")
    private Double price = 0.0;

    @Column(name = "discount")
    private Double discount = 0.0;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category = new Category();

    @ManyToOne
    @JoinColumn(name = "credit_card_id")
    private CreditCard creditCard = new CreditCard();

    public Purchase(Long id, String name, String locale, String quantiOrKilo, int typeProduct, String date, Double price, Double discount) {
        this.id = id;
        this.name = name;
        this.locale = locale;
        this.quantiOrKilo = quantiOrKilo;
        this.typeProduct = typeProduct;
        this.date = date;
        this.price = price;
        this.discount = discount;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLocale() {
        return locale;
    }

    public void setLocale(String locale) {
        this.locale = locale;
    }

    public String getQuantiOrKilo() {
        return quantiOrKilo;
    }

    public void setQuantiOrKilo(String quantiOrKilo) {
        this.quantiOrKilo = quantiOrKilo;
    }

    public int getTypeProduct() {
        return typeProduct;
    }

    public void setTypeProduct(int typeProduct) {
        this.typeProduct = typeProduct;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public Double getDiscount() {
        return discount;
    }

    public void setDiscount(Double discount) {
        this.discount = discount;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

    public CreditCard getCreditCard() {
        return creditCard;
    }

    public void setCreditCard(CreditCard creditCard) {
        this.creditCard = creditCard;
    }
}

File: myshoppinglistapi/src/main/kotlin/com/example/myshoppinglistapi/entities/Purchase.kt
package com.example.myshoppinglistapi.entities

import com.example.myshoppinglistapi.utils.TypeFrequencyRepeat
import javax.persistence.*

@Entity
@Table(name = "purchases")
class Purchase() {

    @field:Id
    @field:GeneratedValue(strategy = GenerationType.IDENTITY)
    @field:Column(name = "id_purchase")
    var id: Long = 0L

    @field:Column(name = "name")
    var name: String = ""

    @field:Column(name = "locale")
    var locale: String = ""

    @field:Column(name = "amount_or_kilo")
    var amount_or_kilo: String = ""

    @field:Column(name = "type_product")
    var type_product: Int = 0

    @field:Column(name = "date")
    var date: String = "24-01-2022"

    @field:Column(name = "price")
    var price: Double = 0.0

    @field:Column(name = "discount")
    var discount: Double = 0.0

    @ManyToOne
    @JoinColumn(name = "category_id")
    var category: Category = Category()

    @ManyToOne
    @JoinColumn(name = "credit_card_id")
    var creditCard: CreditCard = CreditCard()

    @field:Column(name = "is_repeat")
    var is_repeat: Boolean = false

    @field:Column(name = "date_repeat")
    var date_repeat: String = ""

    @field:Column(name = "frequency_repeat")
    var frequency_repeat: TypeFrequencyRepeat = TypeFrequencyRepeat.NEVER

    @field:Column(name = "repeat_purchase_date")
    var repeat_purchase_date: String = ""

    constructor(
        id: Long,
        name: String,
        locale: String,
        quantiOrKilo: String,
        typeProduct: Int,
        date: String,
        price: Double,
        discount: Double,
        isRepeat: Boolean,
        dateRepeat: String,
        frequencyRepeat: TypeFrequencyRepeat,
        repeatPurchaseDate: String
    ) : this() {
        this.id = id
        this.name = name
        this.locale = locale
        this.amount_or_kilo = quantiOrKilo
        this.type_product = typeProduct
        this.date = date
        this.price = price
        this.discount = discount
        this.is_repeat = isRepeat
        this.date_repeat = dateRepeat
        this.frequency_repeat = frequencyRepeat
        this.repeat_purchase_date = repeatPurchaseDate
    }


}